### DISCUSSION TITLE
License of assets

### DISCUSSION THREAD

**Discussion by loleg** [cables-gl/cables_docs#708] [Answers & Questions]
As a content creator, I apply the same license to my entire project. When reusing other people's works, typically in the form of an embedded texture or audio file, I would like to specify the license of that asset on an individual basis. 

Licenses are typically mentioned in the patch description. It would be better if this metadata was present and visible at a more granular level of the Files editor and instance(s) of the asset. It would avoid people inadvertently copying content that is not open, or not compatible with their license - when, for example, cloning or copying out parts of a patch.

Also, the guidance on asset licensing is currently minimal, and could be expanded to cover other kinds of remixing in the [user license docs](https://cables.gl/docs/faq/licence_payment/userlicence/userlicence)

**Comment by steam0r** [comment]
hey,

you raise a fair point. we just do not really think it's up to us to make sure that people pick the right licence, use the right assets or "behave" when cloning a patch.

we mention the licence on the patchpage, while cloning the op, when showing the patch summary in the editor and also a lot of licence information is being put into any exported patch, including crediting the authors of userops, all the libraries used and some general licence information.

most of these places have an easy explanation of what is allowed and what isn't next to them, or a link away. putting this information (or even only the possiblity to pick a licence) for each asset would be a long list to present on export...

...and then, what is going to happen? are people going to publish patches under "public domain" and have individual assets be "all rights reserved"? what are we supposed to do then, how do we present this to the user?

we are really sorry, and we take this matter really serious (as you might see by the efforts that we already put in place), but in the end we cannot really enforce these licences anyhow and have to hope people "behave"...which in general they do.

feel free to contact me on discord, if you have more questions or someone disrespecting your copyright and i will see what we can do to help you. i hope you understand this...

best,
steam
